lightning-persister/src/lib.rs

```diff
@@ -0,0 +1,395 @@
+			if txid.is_err() { return Err(ChannelMonitorUpdateErr::PermanentFailure); }
+
+			let index = filename.unwrap().split_at(65).1.split('.').next().unwrap().parse();
+			if index.is_err() { return Err(ChannelMonitorUpdateErr::PermanentFailure); }
+
+			let contents = fs::read(&file.path());
+			if contents.is_err() { return Err(ChannelMonitorUpdateErr::PermanentFailure); }
+
+			if let Ok((_, loaded_monitor)) =
+				<(BlockHash, ChannelMonitor<ChanSigner>)>::read(&mut Cursor::new(&contents.unwrap())) {
+				res.insert(OutPoint { txid: txid.unwrap(), index: index.unwrap() }, loaded_monitor);
+			} else {
+				return Err(ChannelMonitorUpdateErr::PermanentFailure);
+			}
+		}
+		Ok(res)
+	}
+}
+
+impl<ChanSigner: ChannelKeys + Readable + Writeable + Send + Sync> channelmonitor::Persist<ChanSigner> for FilesystemPersister {
+	fn persist_new_channel(&self, funding_txo: OutPoint, monitor: &ChannelMonitor<ChanSigner>) -> Result<(), ChannelMonitorUpdateErr> {
+		self.write_channel_data(funding_txo, monitor)
+		  .map_err(|_| ChannelMonitorUpdateErr::PermanentFailure)
+	}
+
+	fn update_persisted_channel(&self, funding_txo: OutPoint, _update: &ChannelMonitorUpdate, monitor: &ChannelMonitor<ChanSigner>) -> Result<(), ChannelMonitorUpdateErr> {
+		self.write_channel_data(funding_txo, monitor)
+		  .map_err(|_| ChannelMonitorUpdateErr::PermanentFailure)
+	}
+}
+
+impl Drop for FilesystemPersister {
+	fn drop(&mut self) {
+		fs::remove_dir_all(&self.path_to_channel_data).unwrap();
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	extern crate lightning;
+	extern crate bitcoin;
+	use crate::FilesystemPersister;
+	use bitcoin::blockdata::block::{Block, BlockHeader};
+	use bitcoin::hashes::hex::FromHex;
+	use bitcoin::Txid;
+	use DiskWriteable;
+	use Error;
+	use lightning::chain::transaction::OutPoint;
+	use lightning::{check_closed_broadcast, check_added_monitors};
+	use lightning::ln::features::InitFeatures;
+	use lightning::ln::functional_test_utils::*;
+	use lightning::ln::msgs::ErrorAction;
+	use lightning::util::enforcing_trait_impls::EnforcingChannelKeys;
+	use lightning::util::events::{MessageSendEventsProvider, MessageSendEvent};
+	use lightning::util::ser::Writer;
+	use lightning::util::test_utils;
+	use std::fs;
+	use std::io;
+	#[cfg(not(target_os = "windows"))]
+	use {
+		bitcoin::hashes::sha256::Hash as Sha256,
+		bitcoin::hashes::Hash,
+		lightning::{unwrap_send_err, get_payment_preimage_hash},
+		lightning::ln::channelmanager::{PaymentSendFailure, PaymentPreimage, PaymentHash},
+		lightning::routing::router::get_route,
+		lightning::util::errors::APIError,
+	};
+	#[cfg(target_os = "windows")]
+	use {
+		lightning::get_event_msg,
+		lightning::ln::msgs::ChannelMessageHandler,
+	};
+
+	struct TestWriteable{}
+	impl DiskWriteable for TestWriteable {
+		fn write(&self, writer: &mut fs::File) -> Result<(), Error> {
+			writer.write_all(&[42; 1])
+		}
+	}
+
+	#[test]
+	fn test_filesystem_persister() {
```

Comment:
If you can add one line about what this test is checking "Test after few relayed payments and a channel force-close that persisted have been updated X times as expected"